mod/execution/pkg/client/engine.go

```diff
@@ -149,16 +150,16 @@ func (s *EngineClient[
 
 	// Call and check for errors.
 	result, err := s.Client.GetPayload(cctx, payloadID, forkVersion)
-	switch {
-	case err != nil:
+	if err != nil {
 		if errors.Is(err, engineerrors.ErrEngineAPITimeout) {
 			s.metrics.incrementGetPayloadTimeout()
 		}
 		return result, s.handleRPCError(err)
-	case result == nil:
+	}
+	if result == nil {
```

Comment:
note: this checks and the others like this in this file are redundant. Not sure we want to drop them, nor that in case I should do it in this PR. Prolly will open a separate one